var/spack/repos/builtin/packages/elemental/package.py

```diff
@@ -62,15 +65,21 @@ class Elemental(CMakePackage):
     variant('build_type', default='Release',
             description='The build type to build',
             values=('Debug', 'Release'))
+    variant('blas', default='openblas', values=('openblas', 'mkl'),
+            description='Enable the use of OpenBlas/MKL')
```

Comment:
I think that that way would be great, but my understanding is that virtual dependencies are unable to properly forward their variants and so when Elemental is built with options that have to be forwarded to the BLAS library this mechanism doesn't work.  I believe that this is noted in bug 1712.  I was previously advised to make the dependencies explicit until this bug was resolved.